src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -395,21 +400,22 @@ public void execute(BuildScanExtension buildScan) {
                 buildScan.value("Git status", gitStatus);
             }
 
-            if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-                if (gitRepo.contains("github.com/") || gitRepo.contains("github.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)github\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("Github source", "https://github.com/" + repoPath + "/tree/" + gitCommitId);
-                    }
-                } else if (gitRepo.contains("gitlab.com/") || gitRepo.contains("gitlab.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)gitlab\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("GitLab Source", "https://gitlab.com/" + repoPath + "/-/commit/" + gitCommitId);
+            Optional<String> gitHubUrl = envVariable("GITHUB_SERVER_URL", providers);
+            Optional<String> gitHubRepository = envVariable("GITHUB_REPOSITORY", providers);
+            if (gitHubUrl.isPresent() && gitHubRepository.isPresent()) {
+                buildScan.link("GitHub source", gitHubUrl.get() + "/" + gitHubRepository.get() + "/tree/" + gitCommitId);
+            } else if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
+                try {
+                    URI gitRepoUri = new URI(gitRepo);
+                    String gitRepoHost = gitRepoUri.getHost();
+                    String gitRepoPath = gitRepoUri.getPath().endsWith(".git") ? gitRepoUri.getPath().substring(0, gitRepoUri.getPath().length() - 4) : gitRepoUri.getPath();
+                    if (gitRepo.contains("github")) {
```

Comment:
Under what circumstances would GITHUB_SERVER_URL and GITHUB_REPOSITORY not be present but it's still a repository with the name GitHub in it? A local build?

Reply:
Correct - for local builds we would not expect those environment variables to be present, but if the remote repository is hosted on a GitHub server, it may have "github" in the hostname.

Reply:
Got it.